features/call/src/main/kotlin/io/element/android/features/call/ui/CallScreenPresenter.kt

```diff
@@ -166,6 +175,36 @@ class CallScreenPresenter @AssistedInject constructor(
         }
     }
 
+    @Composable
+    private fun HandleMatrixClientSyncState() {
+        val coroutineScope = rememberCoroutineScope()
+        DisposableEffect(Unit) {
+            val client = (callType as? CallType.RoomCall)?.sessionId?.let {
+                matrixClientsProvider.getOrNull(it)
+            } ?: return@DisposableEffect onDispose { }
+
+            coroutineScope.launch {
+                client.syncService().syncState
+                    .onEach { state ->
+                        if (state != SyncState.Running) {
+                            client.syncService().startSync()
+                        }
+                    }
+                    .collect()
+            }
+            onDispose {
+                // We can't use the local coroutine scope here because it will be disposed before this effect
+                appCoroutineScope.launch {
+                    client.syncService().run {
+                        if (syncState.value == SyncState.Running) {
+                            stopSync()
```

Comment:
Is there no risk of stopping the sync after it's been restarted inside `LoggedInFlowNode`?

Reply:
`LoggedInFlowNode` already tracks the current state and tries to restart the flow automatically in `LoggedInFlowNode.observeSyncStateAndNetworkStatus`, similar to what I do here.

Reply:
By the way, what happens if the network is lost? (airplane)

Reply:
Oh, nice catch. Endless loops of syncService starting.

Reply:
I guess this code could be extracted an reusable : 
```
combine(
                    // small debounce to avoid spamming startSync when the state is changing quickly in case of error.
                    syncService.syncState.debounce(100),
                    networkMonitor.connectivity
                ) { syncState, networkStatus ->
                    Pair(syncState, networkStatus)
                }
                    .collect { (syncState, networkStatus) ->
                        Timber.d("Sync state: $syncState, network status: $networkStatus")
                        if (syncState != SyncState.Running && networkStatus == NetworkStatus.Online) {
                            syncService.startSync()
                        }
                    }
```

Reply:
For now I just copied it, we'll see how we can extract this to some other component or extension function.